src/azul/indexer/__init__.py

```diff
@@ -200,6 +200,31 @@ def __str__(self) -> str:
     def effective(self) -> SOURCE_SPEC:
         return attr.evolve(self, prefix=self.prefix.effective)
 
+    def contains(self, other: 'SourceSpec') -> bool:
+        """
+        >>> p = SimpleSourceSpec.parse
+
+        >>> p('foo:4').contains(p('foo:42'))
+        True
+
+        >>> p('foo:42').contains(p('foo:4'))
+        False
+
+        >>> p('foo:42').contains(p('foo:42'))
+        True
+
+        >>> p('foo:1').contains(p('foo:2'))
+        False
+
+        >>> from azul.terra import TDRSourceSpec
+        >>> TDRSourceSpec.parse('tdr:foo:snapshot/bar:2').contains(p('foo:2'))  # doctest: +IGNORE_EXCEPTION_DETAIL
+        Traceback (most recent call last):
+        ...
+        AssertionError:
+        """
+        assert isinstance(other, type(self)), (self, other)
+        return other.prefix.common.startswith(self.prefix.common)
```

Comment:
```suggestion
        assert isinstance(other, SourceSpec)
        return isinstance(other, type(self)) and other.prefix.common.startswith(self.prefix.common
```

And no test case for the assert failing. Might need to wrap the return statement.